paddle/fluid/framework/ir/mkldnn/cpu_quantize_squash_pass.cc

```diff
@@ -104,51 +104,24 @@ void CPUQuantizeSquashPass::FindNodesToKeep(
   AddStatis(found_count);
 }
 
-bool CPUQuantizeSquashPass::IsDequantizeInputUint8(
-    const Node* dequant_in) const {
-  PADDLE_ENFORCE_EQ(
-      dequant_in->inputs.size(),
-      1,
-      platform::errors::InvalidArgument(
-          "Dequantize (id: %f) should have only one input.", dequant_in->id()));
-  if (dequant_in->inputs[0]->IsOp()) {
-    auto prev_op = dequant_in->inputs[0]->Op();
-    std::string act_name;
-    if (prev_op->Type() == "relu") {
-      return true;
-    } else {
-      if (prev_op->Type() == "conv2d") {
-        act_name = "fuse_activation";
-      } else if (prev_op->Type() == "fc") {
-        act_name = "activation_type";
-      }
-      if (!act_name.empty()) {
-        auto act = prev_op->GetAttrIfExists<std::string>(act_name);
-        if (act == "relu" || act == "relu6") {
-          return true;
-        }
-      }
-    }
-  }
-  return false;
-}
-
 bool CPUQuantizeSquashPass::IsDequantizeQuantizeIncompatible(
-    Node* quant_op, Node* dequant_in, Node* next_op) const {
+    Node* quant_op, Node* dequant_op, Node* next_op) const {
   bool is_concat_signed =
       quant_op->Op()->GetAttrIfExists<bool>("is_negative_input");
-  bool is_input_unsigned = IsDequantizeInputUint8(dequant_in);
+  bool is_input_signed =
+      dequant_op->Op()->GetAttrIfExists<bool>("is_negative_input");
+
   /* TODO(sfraczek): remove elementwise from this condition when BinaryMKLDNN
    kernel will support two different input data types */
   bool is_next_op_concat_or_elementwise =
       next_op->Op()->Type() == "concat" ||
       next_op->Op()->Type().find("elementwise") == 0;
-  if (is_next_op_concat_or_elementwise && is_concat_signed &&
-      is_input_unsigned) {
+  if (is_next_op_concat_or_elementwise &&
```

Comment:
Whether there is a float type input in elementwise, concat or fc, I think the judgment of this case should be added.

Reply:
I'm not sure if I understand. Because at this stage, input to quantize is always FP32. We only remove connections here if we have `(int8/uint8)->dequantize->(fp32)->quantize->(int8/uint8)`. When we have `op->(fp32)->quantize->(int8/uint8)` we leave it as is.

Reply:
![image](https://user-images.githubusercontent.com/30516196/191740593-1976db4c-a16e-4ac1-b83c-04e373cbe602.png)
I'm talking about situations like this

Reply:
Oh, such a situation should not take place, because in `cpu_quantize_pass`, we add quantizes before all concat inputs

Reply:
ok I got it.